src/coreComponents/constitutive/solid/porosity/BiotPorosity.hpp

```diff
@@ -89,13 +92,74 @@ class BiotPorosityUpdates : public PorosityBaseUpdates
   void updateBiotCoefficient( localIndex const k,
                               real64 const bulkModulus ) const
   {
-    m_biotCoefficient[k] = 1 - bulkModulus / m_grainBulkModulus;
+    m_biotCoefficient[k] = 1.0 - bulkModulus / m_grainBulkModulus;
+  }
+
+  /**
+   * @brief update transversely isotropic Biot's tensor.
+   * @param c11 (1,1) component of the stiffness matrix
+   * @param c12 (1,2) component of the stiffness matrix
+   * @param c13 (1,3) component of the stiffness matrix
+   * @param c33 (3,3) component of the stiffness matrix
+   */
+  GEOSX_HOST_DEVICE
+  void updateBiotTensor( localIndex const k,
+                         real64 const c11,
+                         real64 const c12,
+                         real64 const c13,
+                         real64 const c33 ) const
+  {
+    // Grain compliance tensor (isotropic grain is assumed)
+    real64 grainPoissonRatio = conversions::BulkModAndShearMod::
+                                 toPoissonRatio( m_grainBulkModulus, m_grainShearModulus );
+
+    real64 grainYoungModulus = conversions::BulkModAndShearMod::
+                                 toYoungsMod( m_grainBulkModulus, m_grainShearModulus );
+
+    real64 s11Grain = 1.0 / grainYoungModulus;
+    real64 s12Grain = -grainPoissonRatio / grainYoungModulus;
+    real64 s13Grain = s12Grain;
+    real64 s33Grain = s11Grain;
+
+    // Components of Biot's matrix
+    m_b11[k] = 1.0 - ( s11Grain + s12Grain ) * ( c11 + c12 )
+               - s13Grain * ( c11 + c12 + 2.0 * c13 )
+               - s33Grain * c13;
+
+    m_b33[k] = 1.0 - 2.0 * s11Grain * c13
+               - 2.0 * s12Grain * c13
+               - 2.0 * s13Grain * ( c13 + c33 )
+               - s33Grain * c33;
+
+    m_b22[k] = m_b11[k];
+  }
+
+  /**
+   * @brief Biot's coefficient is approximated by the average of the Biot's tensor diagonal.
+   */
+  GEOSX_HOST_DEVICE
+  void updateBiotCoefficient( localIndex const k,
+                              real64 const c11,
+                              real64 const c12,
+                              real64 const c13,
+                              real64 const c33 ) const
+  {
+    updateBiotTensor( k, c11, c12, c13, c33 );
+
+    m_biotCoefficient[k] = ( m_b11[k] + m_b22[k] + m_b33[k] ) / 3.0;
   }
 
 protected:
+
   arrayView1d< real64 > m_biotCoefficient;
 
   real64 m_grainBulkModulus;
+
+  real64 m_grainShearModulus;
+
+  arrayView1d< real64 > m_b11; ///< (1,1) element of TI Biot's matrix
+  arrayView1d< real64 > m_b22; ///< (2,2) element of TI Biot's matrix
+  arrayView1d< real64 > m_b33; ///< (3,3) element of TI Biot's matrix
```

Comment:
this biot's matrix is only used to compute the biotCoefficient. I don't think we need to store it.

Reply:
@CusiniM For transversely isotropic or orthotropic rock, we should use the anisotropic Biot's tensor to update stress and porosity